### PR TITLE
SC-9018; SC-9003 - replaces usage of api /schools enpoint with schoolsList

### PR DESCRIPTION
**Pull request by virgilchiriac**
# Description

## Links to Tickets or other pull requests
https://ticketsystem.schul-cloud.org/browse/SC-9003
https://github.com/hpi-schul-cloud/schulcloud-server/pull/2504

## Approval for review
- [x] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

> Notice: Please remove the WIP label if the PR is ready to review, otherwise nobody will review it.

### Link to Definition of Done
More and detailed information on the *definition of done* can be found [on Confluence](https://docs.schul-cloud.org/pages/viewpage.action?pageId=92831762)

### CODE REVIEW

File: controllers/login.js [hunk 1]
```diff
@@ -133,13 +133,9 @@ router.all('/login/superhero/', (req, res, next) => {
...
-const ssoSchoolData = (req, systemId) => api(req).get('/schools/', {
-	qs: {
-		systems: systemId,
-	},
-}).then((schools) => {
-	if (schools.data.length > 0) {
-		return schools.data[0];
+const ssoSchoolData = (req, systemId) => api(req).get('/schoolsList').then((schools) => {
+	if (schools.length > 0) {
+		return schools.find((school) => school.systems.find((system) => system._id === systemId));
```
Comment (CeEv) [inline controllers/login.js:138]: Can we pass the requested systemId as optional query to backend to get the reduced list directly?

File: controllers/login.js [hunk 2]
```diff
@@ -133,13 +133,9 @@ router.all('/login/superhero/', (req, res, next) => {
...
 	}
 	return undefined;
 }).catch(() => undefined); // TODO: fixme this is a very bad error catch
```
Comment (CeEv) [inline controllers/login.js:141]: In error case we get no information. Can we log an error and return the 2 retun undefined with empty array? 
We must only replace the line 168 if (schools.length > 0)

File: views/authentication/forms/login.hbs [hunk 3]
```diff
@@ -46,9 +46,9 @@
 						<select id="school" name="schoolId" class="school search-enabled" data-placeholder="{{$t "login.input.school"}}">
 								<option selected value="">{{$t "login.input.noSchoolsSelection"}}</option>
 							{{#each schools}}
-								<option value="{{this._id}}">{{this.name}}</option>
+								<option value="{{this._id}}" data-systems='{{{json this.systems}}}'>{{this.name}}</option>
 							{{else}}
-								<option disabled>{{$t "login.input.noSchoolsAvailable"}}</option>
+								<option disabled>{{$t "login.input.noSchoolsAvailable"}}</opJSON.stringify(ttion>
```
Comment (CeEv) [inline views/authentication/forms/login.hbs:51]: </opJSON.stringify(ttion>  typo ?

**Review by CeEv — CHANGES_REQUESTED** [review]

File: controllers/login.js (hunk 2 above)
Comment (virgilchiriac) [reply controllers/login.js]: I rewrote this part. Should be more simple, the code is moved to line 160. There we request from cache the list of all schools, to check if the school and systemId match with the one from result.

File: controllers/login.js (hunk 1 above)
Comment (virgilchiriac) [reply controllers/login.js]: My idea was to have an endpoint which is as simple as possible, with no params, no extra api calls needed.
This is only used from login, and having the cache (increased form 1 to 30 minutes now), it should be performant.

**Review by CeEv — APPROVED** [review]

**Pull request merged**